src/components/CampsiteListItem.js

```diff
@@ -22,64 +24,97 @@ const renderIcons = ({features, facilities}) => {
-    const {campsite_form: {accessibility_options}} = campsite;
-    const {id, title, description, nearestTown, accessibility, siteImageData, features, facilities} = site;
-    const preparedDescription = description && description.length > 57 ?  `${description.substring(0, 57)}...` : description;
-
-    return (
-        <ListItem
-            bottomDivider={true}
-            onPress={() => getSiteDetail({selectedSite: site, navigate})}
-            leftAvatar={{avatarStyle: siteAvatarContainerStyle, source: siteImageData ? {uri: `data:image/png;base64,${siteImageData}`} : require('../../assets/starTent.jpg')}}
-            title={
-                <View style={titleContainerStyle}>
-                    <Text style={titleView}>{title}</Text>
-                    <Badge
-                        wrapperStyle={badgeWrapperStyle}
-                        containerStyle={badgeContainerStyle}
-                    >
-                        <Text style={nearestTownStyle}>{nearestTown}</Text>
-                    </Badge>
-                    <View style={subtitleView}>
-                        <Text>{preparedDescription}</Text>
-                    </View>
-                    <View style={subtitleView}>
+class CampsiteListItem extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            siteImageData: require(`../../assets/defaultSiteImage.jpg`)
+        }
+    }
 
-                        <Text style={accessibilityStyle}>{accessibility_options[accessibility]}</Text>
+    replaceImageData = () => {
+        returnImageForSiteKey({siteKey: this.props.site.id})
+            .then(imageData => {
+                this.setState({
+                    siteImageData: {uri: `data:image/png;base64,${imageData}`}
+                })
+            });
+    };
+
+    render() {
+        const {mainContainerStyle, contentContainerStyle, textStyle, IconContainer, subtitleView, titleContainerStyle, siteAvatarContainerStyle, titleView, nearestTownStyle, badgeWrapperStyle, badgeContainerStyle, accessibilityStyle} = styles
+        const {campsite_form: {accessibility_options}} = campsite;
+        const {title, description, nearestTown, accessibility, features, facilities} = this.props.site;
+        const preparedDescription = description && description.length > 57 ? `${description.substring(0, 57)}...` : description;
+
+        return (
+            <ListItem
+                containerStyle={mainContainerStyle}
+                titleStyle={contentContainerStyle}
+                bottomDivider={true}
+                onPress={() => this.props.getSiteDetail({selectedSite: this.props.site, navigate: this.props.navigate})}
+                leftAvatar={
+                    <Image
+                        source={this.state.siteImageData}
+                        style={siteAvatarContainerStyle}
+                        onLoadStart={this.replaceImageData}
```

Comment:
Great native call that all `Image`s have, is variations of `onLoad` which we can call to make things happen. What we do here is have the `source={this.state.siteImageData}` which is the placeholder image, and then `onLoadStart={this.replaceImageData}` and when the `AsyncStorage` data comes back, we update `this.state.siteImageData` and the source of the image changes.